File: Semester_1/Labs/Lab6/selection_sort.cpp
#include <iostream>

void selectionSort(int a[]){
    int length = sizeof(a) / sizeof(a[0]);
    for (int i = 0; i < length - 1; i++) {
        int minIndex = i;
        for (int j = i + 1; j < length; j++) {
            if (a[j] < a[minIndex])
                minIndex = j;
        std::swap(a[i], a[minIndex]);
        }
    }
}

File: Semester_1/Labs/Lab6/selection_sort.cs
public class selection_sort
{
    public void sort(int[] a)
    {
        for (int i = 0; i < a.Length; i++)
        {
            var minIndex = i;
            for (int j = i; j < a.Length; j++)
            {
                if (a[minIndex]>a[j])
                {
                    minIndex = j;
                }
                
            }

            if (minIndex!=i)
            {
                (a[i], a[minIndex]) = (a[minIndex], a[i]);
            }
        }
    }
}